### DISCUSSION TITLE
Prisma generates the wrong SQL query for filter JSON

### DISCUSSION THREAD

**Discussion by felinto-dev** [prisma/prisma#9743] [Q&A] [answered]
I need help filtering values ​​from a table that is in JSON format. I am using the last available version of PostgreSQL.

![Screenshot_973](https://user-images.githubusercontent.com/32253743/137122291-4d368b1e-65de-436d-b56e-da00d99789f9.png)

I'm trying to filter from this JSON.
I need to find any value in the table where the JSON "hottok" key is equal to "jSFVVN4QyWqHp6VFEbfVCIypstKNy1d66c4006-e559-43ee-8e61-96ef8298df30"

Here is the JSON that was shown in the print for ease:
{ "hottok": "jSFVVN4QyWqHp6VFEbfVCIypstKNy1d66c4006-e559-43ee-8e61-96ef8298df30" }

When using Prisma it generated the following Query:

`SELECT "public".."PlatformAccount"."id", "public"."PlatformAccount"."nickname", "public"."PlatformAccount"."authorizationData", "public"."PlatformAccount"."platform", " public"."PlatformAccount"."token", "public"."PlatformAccount"."ownerId", "public"."PlatformAccount"."createdAt", "public"."PlatformAccount"."updatedAt" FROM "public" ."PlatformAccount" WHERE "public"."PlatformAccount"."authorizationData"::jsonb = '{"path":["hottok"],"equals":"jSFVVN4QyWqHp6VFEbfVCIypstKNy1d66c4006-e559-43ee-8df61-96"ef 1 OFFSET 0`

However, when I run it against the database I don't get any results. I would like help to know what is the problem with this query and how I can adjust it to achieve the desired result.

PS: I am not sure if this is a bug with Prisma because I am not SQL expert. I am following the official documentation about it:
https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filter-on-object-property

![Screenshot_975](https://user-images.githubusercontent.com/32253743/137122939-cdba3d6c-cca8-47a3-96df-38db93824527.png)

**Comment by ryands17** [comment, marked as answer]
@felinto-dev 👋 
 It should be like this (the outer `equals` should be removed):

```ts
await prisma.platformAccount.findFirst({
    where: {
      authorisationData: {
        path: ['hottok'],
        equals: 'something',
      },
    },
})
```

**Reply by felinto-dev** [reply]
I can't believe that I don't look for it!
Unfortunately, I get some problems with Typescript types

any clue?
![Screenshot_977](https://user-images.githubusercontent.com/32253743/137127738-8e78b6e2-e7df-4f29-8828-59b0ef10de32.png)
![Screenshot_978](https://user-images.githubusercontent.com/32253743/137127745-64cd0ed9-cb34-41ac-8f9e-bb6c0a409b79.png)

**Reply by ryands17** [reply]
Have you added the `filterJson` preview feature as shown [here](https://www.prisma.io/docs/concepts/components/preview-features/client-preview-features/)?

**Reply by felinto-dev** [reply]
No!
The error is entirely mine, however, the @prisma typescript types led me to the mistake. I think maybe it's a point to improve. It helped a lot! It was so simple! Thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
